app/Model/Book.php

```diff
@@ -41,12 +42,14 @@ class Book extends Model
     ];
 
     /**
+     * Relationship hasMany with Post
      * Relationship morphMany with Post
      *
      * @return array
     */
     public function posts()
     {
+        return $this->hasMany(Post::class);
```

Comment:
xoá return $this->hasMany(Post::class); đi
trong table post chỉ có các trường của morph `(postable_id, postable_type)` chứ k có trường `book_id`
nên phải dùng $this->morphMany(Post::class, 'postable');